transport/http/outbound.go

```diff
@@ -135,7 +135,12 @@ func (t *Transport) NewOutbound(chooser peer.Chooser, opts ...OutboundOption) *O
 	client := t.client
 	if o.tlsConfig != nil {
 		client = createTLSClient(o)
-		o.urlTemplate.Scheme = "https"
+		// Create a copy of the url template to avoid scheme changes impacting
+		// other outbounds as the base url template is shared across http
+		// outbounds.
+		ut := *o.urlTemplate
+		ut.Scheme = "https"
+		o.urlTemplate = &ut
```

Comment:
this is unfortunate :(
initializing an object with a shared global var is a bad idea. Do you think this might be better and prevents more future issues? 
- remove `defaultURLTemplate` global and initialize it inside `NewOutbound`
something like:
```go
urlTemplate, _ := url.Parse("http://localhost")
if o.tlsConfig != nil { 
    urlTemplate.Scheme = "http"
}
o.urlTemplate = urlTemplate
```

Reply:
That makes sense and I actually did try that earlier. But there is a catch, we have an outbound option that overrides the URL template.
https://github.com/yarpc/yarpc-go/blob/e58c0b765a02bf9866b6b08a9b96f83204f9c812/transport/http/outbound.go#L69

Although the above option creates a new template for each outbound, I still feel copying the URL template and overriding the schema is safer and avoids surprises in the future if somehow the URL template were shared.

Reply:
sure either option works.